### PR TITLE
Neptune Logger Improvements

### PR DESCRIPTION
**Pull request by jakubczakon**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Neptune Logger improvement #908

- added `close_after_fit` argument that lets you leave the logger open after the `trainer.fit(model)` finishes. It makes logging test metrics `trainer.test(model`, images or checkpoints easy via
generic neptune experiment methods `.log_metric`, `.log_image`, `.log_artifact`.
- added "ANONYMOUS" token in examples to let people try it out without registering

### CODE REVIEW

**Comment by jakubczakon** [comment]
@Borda 
I haven't added this option to other loggers as I saw people did some work there already (comet).
I can add this `close_after_fit` option to other loggers but I just don't want to step into other folks territory (where they may not appreciate me being) :). 
It's your call.

**Comment by Borda** [comment]
hey there, we have added GPU CI test, so could we kindly ask to rebase/merge master which will trigger these tests so we do not need to test it manually... Thx for your understanding :robot:

**Comment by jakubczakon** [comment]
Sure, no worries, I'll work on it.

File: pytorch_lightning/loggers/neptune.py [hunk 1]
```diff
@@ -85,40 +87,89 @@ def any_lightning_module_function_or_hook(...):
...
+
+            # When you are done, stop the experiment
+            neptune_logger.experiment.stop()
+
+        You can go and see an example experiment here:
+        https://ui.neptune.ai/o/shared/org/pytorch-lightning-integration/e/PYTOR-66/charts
+
         Args:
-            api_key (str | None): Required in online mode. Neputne API token, found on https://neptune.ml.
+            api_key (str | None): Required in online mode.
```
Comment (Borda) [inline pytorch_lightning/loggers/neptune.py]: drop duplicate arg types (they are already in type hints)

**Comment by jakubczakon** [comment]
I think it's done @Borda

**Pull request merged**